src/hdf5_dl.c

```diff
@@ -97,44 +97,167 @@ hid_t H5E_ERR_CLS_g = -1;
 
 static bool is_init = false;
 
+/*
+ * Try to find a symbol within a library
+ * 
+ * handle: Handle to the library
+ * symbol: Symbol to look for
+ * Returns: a pointer to the symbol or NULL 
+ * if the symbol can't be found 
+ */
+void *find_sym(void *handle, const char *symbol) {
+
+  void *ret = NULL, *err = NULL;
+  dlerror(); /* clear error code */
+  ret = dlsym(handle, symbol);
+
+  if(ret != NULL && (err = dlerror()) == NULL)
+    return ret;
+  else
+    return NULL;
+}
 
 /* Initialize the dynamic loading of symbols and register the plugin
  *
  * libname: Name of the DLL from which to load libHDF5 symbols
  * Returns: a value < 0 if an error occured
  */
-int init_filter(const char* libname)
+int init_filter(const char *libname)
 {
     int retval = -1;
-  	void * handle;
+  	void *handle = NULL, *symbol = NULL;
```

Comment:
The meaning of the return value in this function is inconsistent with the documentation, although it looks like this was the same in the previous implementation, i.e. you return -1 if all symbols have been successfully resolved, which I would consider a success, but according to the docstring (and C conventions) should indicate a failure.